src/Unread.ts

```diff
@@ -57,14 +59,21 @@ export function doesRoomHaveUnreadMessages(room: Room): boolean {
     // despite the name of the method :((
     const readUpToId = room.getEventReadUpTo(myUserId);
 
-    // as we don't send RRs for our own messages, make sure we special case that
-    // if *we* sent the last message into the room, we consider it not unread!
-    // Should fix: https://github.com/vector-im/element-web/issues/3263
-    //             https://github.com/vector-im/element-web/issues/2427
-    // ...and possibly some of the others at
-    //             https://github.com/vector-im/element-web/issues/3363
-    if (room.timeline.length && room.timeline[room.timeline.length - 1].getSender() === myUserId) {
-        return false;
+    if (!SettingsStore.getValue("feature_thread")) {
+        // as we don't send RRs for our own messages, make sure we special case that
+        // if *we* sent the last message into the room, we consider it not unread!
+        // Should fix: https://github.com/vector-im/element-web/issues/3263
+        //             https://github.com/vector-im/element-web/issues/2427
+        // ...and possibly some of the others at
+        //             https://github.com/vector-im/element-web/issues/3363
+        if (room.timeline.length && room.timeline[room.timeline.length - 1].getSender() === myUserId) {
+            return false;
+        }
+    } else {
+        const threadState = RoomNotificationStateStore.instance.getThreadsRoomState(room);
+        if (threadState.color > 0) {
+            return true;
+        }
```

Comment:
This change feels like it may regress those 3 linked issues

Also I would have thought you'd want the RoomNotificationState here as that wraps the ThreadsRoomNotificationState and seemingly the `ThreadsRoomNotificationState` only considers threads and not the room main timeline?

Reply:
So the 3 issues assume a single timeline... which is not true anymore.
It would exit fast if the last message in the main timeline was from you even though there's a new message in a thread. 

Well, the `ThreadsRoomNotificationState` tells you whether there's at least one unread message in a thread. And the logic that goes below will check the main room timeline.

When this function is executed, `RoomNotificationState` has not been updated yet, as it's waiting on this function to be resolved.